environments/common/src/main/java/org/jboss/weld/environment/deployment/discovery/AbstractDiscoveryStrategy.java

```diff
@@ -47,37 +47,63 @@
  * @author Matej Briškár
  * @author Martin Kouba
  * @author Jozef Hartinger
+ * @author <a href="https://about.me/lairdnelson"
+ * target="_parent">Laird Nelson</a>
  */
 public abstract class AbstractDiscoveryStrategy implements DiscoveryStrategy {
 
-    protected final ResourceLoader resourceLoader;
+    protected ResourceLoader resourceLoader;
 
-    protected final Bootstrap bootstrap;
+    protected Bootstrap bootstrap;
 
-    protected final Set<Class<? extends Annotation>> initialBeanDefiningAnnotations;
+    protected Set<Class<? extends Annotation>> initialBeanDefiningAnnotations;
 
     protected BeanArchiveScanner scanner;
 
     private final List<BeanArchiveHandler> handlers;
 
+    protected AbstractDiscoveryStrategy() {
+        this.handlers = new LinkedList<BeanArchiveHandler>();
+    }
+
     /**
      *
      * @param resourceLoader
      * @param bootstrap
      * @param initialBeanDefiningAnnotations
      */
     public AbstractDiscoveryStrategy(ResourceLoader resourceLoader, Bootstrap bootstrap, Set<Class<? extends Annotation>> initialBeanDefiningAnnotations) {
+        this();
```

Comment:
There is no need to call `this()` and use `this` keyword for the setters...

Reply:
OK; I'll remove `this` wherever possible and will duplicate the `handlers` assignment code.

Reply:
Ah sorry, my comment about `this()` was wrong. I thought it's `super()` ;-). For setters it's useless but for handlers assignment it's ok.